analyzers/src/SonarAnalyzer.CSharp/Rules/SillyBitwiseOperation.cs

```diff
@@ -64,5 +84,101 @@ private void CheckBinary(SonarSyntaxNodeReportingContext context, int constValue
             var binary = (BinaryExpressionSyntax)context.Node;
             CheckBinary(context, binary.Left, binary.OperatorToken, binary.Right, constValueToLookFor);
         }
+
+        private void CheckComparison(SonarSyntaxNodeReportingContext context)
+        {
+            var binary = (BinaryExpressionSyntax)context.Node;
+            if (ShouldRaise(context.SemanticModel, binary.Left, binary.Right, binary.Kind(), false))
+            {
+                context.ReportIssue(Diagnostic.Create(BitwiseRule, binary.GetLocation()));
+            }
+
+            if (ShouldRaise(context.SemanticModel, binary.Right, binary.Left, binary.Kind(), true))
+            {
+                context.ReportIssue(Diagnostic.Create(BitwiseRule, binary.GetLocation()));
+            }
+        }
+
+        private bool ShouldRaise(SemanticModel semanticModel, SyntaxNode value, SyntaxNode expectedConstant, SyntaxKind kind, bool constantIsLeft)
+        {
+            if (FindConstant(semanticModel, expectedConstant, Convert.ToDouble) is { } constant
+                && semanticModel.GetSymbolInfo(value).Symbol.GetSymbolType() is { } symbol
+                && Ranges.FirstOrDefault(x => symbol.Is(x.Type)) is { } range)
+            {
+                // Implement out-of-range checks for the types that CS0652 does not.
+                if (symbol.IsAny(FullyImplementedTypes) && (constant < range.MinValue || constant > range.MaxValue))
+                {
+                    return true;
+                }
+
+                // Implement threshold checks for every type.
+                return GetThreshold(kind, constantIsLeft, range) is { } threshold && constant == threshold;
```

Comment:
Note: `constant == threshold` raises a `Bug` on the quality gate, because it's an equality comparison with a double-precision number. 

But for this case, we want to check for exact equality, to see if the constant equals either `T.MinValue` or `T.MaxValue`

Any ideas?

Reply:
I could fix the quality gate with something like this:
```csharp
public static class DoubleExtensions
    {
        const double _3 = 0.001;
        const double _4 = 0.0001;

        public static bool Equals3DigitPrecision(this double left, double right)
        {
            return Math.Abs(left - right) < _3;
        }

        public static bool Equals4DigitPrecision(this double left, double right)
        {
            return Math.Abs(left - right) < _4;
        }
}
```

Do you think this makes sense?